### PR TITLE
docs:(oft-solana): mention amount flag for creation

### PR DESCRIPTION
**Pull request by nazreen**

### CODE REVIEW

File: examples/oft-solana/README.md [hunk 1]
```diff
@@ -198,31 +198,37 @@ sh -c "$(curl -sSfL https://release.solana.com/v1.17.31/install)"
 
 ### Create the Solana OFT
 
-For OFT:
+:information_source: In all the OFT patterns listed below, the SPL token's Mint Authority is set to the **Mint Authority Multisig**, which always has the **OFT Store** as a signer. The multisig is fixed to needing 1 of N signatures.
```
Comment (DanL0) [inline examples/oft-solana/README.md:201]: all except OFT Adapter which is lockbox and doesn't need mint rights

File: examples/oft-solana/README.md [hunk 2]
```diff
@@ -198,31 +198,37 @@ sh -c "$(curl -sSfL https://release.solana.com/v1.17.31/install)"
...
+
+:information_source: For **OFT** and **OFT Mint-And-Burn Adapter**, you have the option to specify additional signers through the `--additional-minters` flag. If you choose not to, you must pass in `--only-oft-store true`, which means only the **OFT Store** will be a signer for the \_Mint Authority Multisig\*.
+
+:warning: If you choose to go with `--only-oft-store`, you will not be able to add in other signers/minters or update the Mint Authority.
```
Comment (DanL0) [inline examples/oft-solana/README.md:205]: Probably good time to add a warning about Freeze Authority as well. If you choose `--only-oft-store` you can't renounce Freeze Authority.

File: examples/oft-solana/README.md [hunk 3]
```diff
@@ -198,31 +198,37 @@ sh -c "$(curl -sSfL https://release.solana.com/v1.17.31/install)"
...
 pnpm hardhat lz:oft:solana:create --eid 40168 --program-id <PROGRAM_ID>
 ```
 
-:warning: You may specify the `--additional-minters` flag to add a CSV of additional minter keys to the mint
-multisig. If you do not want to, you must specify `--only-oft-store true`. If you choose the latter approach, you can never
-substitute in a different mint authority.
+:warning: Use `--additional-minters` flag to add a CSV of additional minter addresses to the Mint Authority Multisig. If you do not want to, you must specify `--only-oft-store true`.
 
-For OFTAdapter:
+:information_source: You can also specify `--amount <AMOUNT>` to have the OFT minted to your deployer address upon token creation.
```
Comment (DanL0) [inline examples/oft-solana/README.md:215]: Can you test this and check if your balance actually increased on deployer account? For me I think I tried it and I wasn't sure if it worked. I'm not sure if it credited my account and created token account.

File: examples/oft-solana/README.md [hunk 4]
```diff
@@ -198,31 +198,37 @@ sh -c "$(curl -sSfL https://release.solana.com/v1.17.31/install)"
...
+#### For OFT Mint-And-Burn Adapter (MABA):
 
 ```bash
 pnpm hardhat lz:oft:solana:create --eid 40168 --program-id <PROGRAM_ID> --mint <TOKEN_MINT> --token-program <TOKEN_PROGRAM_ID>
 ```
 
-:warning: You may specify the `--additional-minters` flag to add a CSV of additional minter keys to the mint
-multisig. If you do not want to, you must specify `--only-oft-store store`. If you choose the latter approach, you can never
-substitute in a different mint authority.
+:warning: Use `--additional-minters` flag to add a CSV of additional minter addresses to the Mint Authority Multisig. If you do not want to, you must specify `--only-oft-store true`.
```
Comment (DanL0) [inline examples/oft-solana/README.md:229]: I think this isn't used for MABA. It seems its only used if !MABA. Maybe we should avoid creating SPL Multisig if someone is creating MABA?
```
if (!isMABA) {
                const { signature } = await transactionBuilder()
                    .add(
                        setAuthority(umi, {
                            owned: mint.publicKey,
                            owner: umiWalletSigner,
                            newAuthority: fromWeb3JsPublicKey(mintAuthorityPublicKey),
                            authorityType: 0,
                        })
                    )
                    .add(
                        setAuthority(umi, {
                            owned: mint.publicKey,
                            owner: umiWalletSigner,
                            newAuthority: fromWeb3JsPublicKey(mintAuthorityPublicKey),
                            authorityType: 1,
                        })
                    )
                    .sendAndConfirm(umi)
                console.log(`setAuthorityTx: ${getExplorerTxLink(bs58.encode(signature), isTestnet)}`)
            }
```

File: examples/oft-solana/README.md [hunk 5]
```diff
@@ -198,31 +198,37 @@ sh -c "$(curl -sSfL https://release.solana.com/v1.17.31/install)"
...
+
+:information_source: You can also specify `--amount <AMOUNT>` to have the OFT minted to your deployer address upon token creation.
```
Comment (DanL0) [inline examples/oft-solana/README.md:231]: This doesn't work for MABA though.

File: examples/oft-solana/README.md [hunk 6]
```diff
@@ -263,6 +269,8 @@ With a squads multisig, you can simply append the `--multisigKey` flag to the en
 
 ### Mint OFT on Solana
 
+This is only relevant for **OFT** and **OFT Mint-And-Burn Adapter** deployments. If you opted to include the `--amount` flag in the create step, that means you already have minted some Solana OFT and you can skip this section.
```
Comment (DanL0) [inline examples/oft-solana/README.md:272]: I think --amount doesn't work for MABA looking at createOFT script and !MABA ifs.

**Review by DanL0 — CHANGES_REQUESTED** [review]

File: examples/oft-solana/README.md (hunk 3 above)
Comment (nazreen) [reply examples/oft-solana/README.md:215]: Ran through the 4 various scenarios. When amount is specified, you can see that Mint/MintTo was called and the OFT is minted to my deployer address

no amount specified, onlyOftStore mode https://solscan.io/tx/3ogRs55ZhjEB7hQbDMXqofqjQQ9xgcXW4EzdpzUYWjgZxTYDhwBTTH7q5wiaG9GQVcK1foYeS4fnAGxZUbqEQhuq?cluster=devnet

amount specified, onlyOftStore mode
https://solscan.io/tx/Lap7SZ1n68djiHaHvZR5RS9akUsqpbWMiRFjqmy7zH5p6M7jiFVVpHsnKqobTWfmoDZTss81gaCSrpPnfj1qopn?cluster=devnet

no amount specified, additionalMinters specified
https://solscan.io/tx/ToVzrSW6GP9RpJwvgUXCkAxWLncMRr32XNLKZmkNEDZRheprrrY39Ra6qqdkxQZh3bpG7yBfzjmKDjbZbNX9X5m?cluster=devnet

amount specified, additionalMinters specified
https://solscan.io/tx/29HAx8YAxSEiuJspsfyEQBec1umcQwJhJ8hPD8bvFDzQd8ZowReBX6L2HBdrMbPdubaCNGsnTQacdP8i4c8NpfFu?cluster=devnet

**Review by DanL0 — APPROVED** [review]
did my own test, seems amount flag works indeed 

```
➜  pnpm hardhat lz:oft:solana:create --eid 40168 --program-id 3ZuKaovZkFLf5StLP2sHDBa65WKR7fjD1AViFugCFqRH --amount 77777777123 --only-oft-store true
bigint: Failed to load bindings, pure JS will be used (try npm run rebuild?)
No additional minters specified.  This will result in only the OFTStore being able to mint new tokens.
created SPL multisig @ DfNUNpuVmLXjAMXdWPsa8gzhyfBuoy26dboCKXZS1GU3
createTokenTx: https://explorer.solana.com/tx/5f4epD2sbjnH9pM1fYKNSuBDNv9TXVbFRZAqkhUis5Z2AFrXPdhMnznieq8iiTUdkTVAvbWQHRUdcvs8zCZQDkeB?cluster=devnet
initOftTx: https://explorer.solana.com/tx/3XRvX4vKuMPbKmCnh6mkcbkfVhgybajebhssA6yBzcvrUMQgfYTKNiKhHQPu1w3Zvfdhr8FbxGTkqssWnegNizF1?cluster=devnet
setAuthorityTx: https://explorer.solana.com/tx/4WcZthNKGb66DLFNag3oJraVyL2TSoZ81sn2B4HS8Na91srHJjXyVXi3azSFEvdHD73hmHzen3cyr6X3yg3Dh8Jv?cluster=devnet
Accounts have been saved to ./deployments/solana-testnet/OFT.json

➜  spl-token balance 4jwaCm6FcyAWR48Yc4YahySSmY8CYzjMfMCx9SY8dyDz
77.777777123
```

**Comment by nazreen** [comment]
> did my own test, seems amount flag works indeed
> 
> ```
> ➜  pnpm hardhat lz:oft:solana:create --eid 40168 --program-id 3ZuKaovZkFLf5StLP2sHDBa65WKR7fjD1AViFugCFqRH --amount 77777777123 --only-oft-store true
> bigint: Failed to load bindings, pure JS will be used (try npm run rebuild?)
> No additional minters specified.  This will result in only the OFTStore being able to mint new tokens.
> created SPL multisig @ DfNUNpuVmLXjAMXdWPsa8gzhyfBuoy26dboCKXZS1GU3
> createTokenTx: https://explorer.solana.com/tx/5f4epD2sbjnH9pM1fYKNSuBDNv9TXVbFRZAqkhUis5Z2AFrXPdhMnznieq8iiTUdkTVAvbWQHRUdcvs8zCZQDkeB?cluster=devnet
> initOftTx: https://explorer.solana.com/tx/3XRvX4vKuMPbKmCnh6mkcbkfVhgybajebhssA6yBzcvrUMQgfYTKNiKhHQPu1w3Zvfdhr8FbxGTkqssWnegNizF1?cluster=devnet
> setAuthorityTx: https://explorer.solana.com/tx/4WcZthNKGb66DLFNag3oJraVyL2TSoZ81sn2B4HS8Na91srHJjXyVXi3azSFEvdHD73hmHzen3cyr6X3yg3Dh8Jv?cluster=devnet
> Accounts have been saved to ./deployments/solana-testnet/OFT.json
> 
> ➜  spl-token balance 4jwaCm6FcyAWR48Yc4YahySSmY8CYzjMfMCx9SY8dyDz
> 77.777777123
> ```

thanks! not sure if you saw my above comment: https://github.com/LayerZero-Labs/devtools/pull/1092#discussion_r1866294582
tested under multiple scenarios

**Pull request merged**